### PR TITLE
fix: #5642 - build deb packages that install on Deb 8

### LINKED ISSUE #5642: Debian - apt repo - package lists or status file could not be parsed or opened
From [Discord](https://discord.com/channels/742820443487993987/746070591097798688/790431615436193792).

```
So moving from filebeat to vector, I noticed that apt would failt on the few (cough) debian8 systems we have left, due to the vector repo... any chance this could get fixed? What I see is this: 
```

```
 ...
Get:2 https://repositories.timber.io jessie/main amd64 Packages [533 B]
Fetched 5,449 B in 3s (1,443 B/s)
Reading package lists... Error!
E: Problem parsing dependency Depends
E: Error occurred while processing vector (NewVersion2)
E: Problem with MergeList /var/lib/apt/lists/repositories.timber.io_public_vector_deb_debian_dists_jessie_main_binary-amd64_Packages
E: The package lists or status file could not be parsed or opened.
```

### PR DESCRIPTION
**Pull request by jamtur01**
This took a lot of fiddly testing. We now hard-code the earliest glibc we support (2.17) as the dependency of the Debian package. This ensures Debian/Ubuntu releases that choke on an empty `Depends:` field. Later Debian/Ubuntu releases should work forwards too.

We've also updated to the latest `cargo-deb` release, which support dropping the `Depends` field if not dependencies are present.

Signed-off-by: James Turnbull <james@lovedthanlost.net>

### CODE REVIEW

File: scripts/environment/prepare.sh [hunk 1]
```diff
@@ -8,7 +8,7 @@ rustup default "$(cat rust-toolchain)"
 rustup component add rustfmt
 rustup component add clippy
 rustup target add wasm32-wasi
-rustup run stable cargo install cargo-deb --version '^1.28.0'
+rustup run stable cargo install cargo-deb --git https://github.com/mmstick/cargo-deb.git
```
Comment (jszwedko) [inline scripts/environment/prepare.sh:11]: Could we pin to a SHA here until the next release? I believe you can use `--rev`.

**Review by jszwedko — APPROVED** [review]
This seems ok, but I'm a little confused. With the changes in `cargo-deb` to drop empty dependencies, do we still need to add a dependency on `libc`?

Maybe we should have had that libc dependency all along though.

**Comment by jamtur01** [comment]
@jszwedko So it is a little confusing... Some history. Previously we had specified `depends = ""` because Cargo-deb's auto-dependency builder populated the libc dependency as the version available in the cross build image (2.31). This was obviously wrong because cross actually supported earlier glibc's (down to 2.2x or so). We then discovered that Centos 7 and Debian 8 had earlier glibc versions (2.17 or so) and that this wouldn't work with binaries generated by cross. So we had folks do the work upstream in cross to support glibc back to 2.17. So technically we didn't need to specify `Depends` until we discovered that earlier Debian and Ubuntu apt commands choke on empty `Depends` fields. Hence the circle closes and we're now explicitly stating the lower end of the glibc support. Distros with earlier glibc versions will still not work (so Debian 7 I suspect and Centos 6) but now the deb ones should fail to install at least for the right reason.

**Comment by jszwedko** [comment]
Gotcha, I see :smile: Thanks for the explanation!

**Pull request merged**